### DISCUSSION TITLE
How to get Model Name/Class in Exception Handler

### DISCUSSION THREAD

**Discussion by TheArKaID** [adonisjs/core#3946] [Help] [answered]
I need to know when 'E_ROW_NOT_FOUND' is thrown, what model is the error coming from, so I can tell User what model/data can't be found.

```js
if (error.code === 'E_ROW_NOT_FOUND') {
  return ctx.response.send({
    'status': 400,
    'message': 'Failed. ' + error.getPreviousClass() + ' Not Found.' // Something like this?
  })
}
```

In Laravel, I could do something like `$e->getPrevious()->getModel()` from `NotFoundHttpException` class.

**Comment by thetutlage** [comment]
It is not possible right now. However, feel free to open a PR in the Lucid repo https://github.com/adonisjs/lucid

**Reply by radiumrasheed** [reply]
I'm looking to raise a PR for this, but will like to understand first, why we chose `Row not found` instead of `${ModelName} not found`.
I see we already pass model name in errors like `E_MISSING_MODEL_ATTRIBUTE` -> `Relation "%s" expects "%s" to exist on "%s" model, but is missing`

___

if there's a valid reason, will it be okay to pass the model in the error class as an extra parameter
like `error.extra.model` or `error.model` ?

**Reply by thetutlage** [reply]
Sticking `error.model` seems okay to me.

**Reply by radiumrasheed** [reply]
I have raised a PR for this. _hopefully, first of many_
https://github.com/adonisjs/lucid/pull/1081

**Comment by radiumrasheed** [comment, marked as answer]
@TheArKaID In the next release, we should be able to use 
`error.model.name`

**Reply by TheArKaID** [reply]
Thanks a lot for addressing this feature
